### ISSUE TITLE
Where can I read your paper?

### ISSUE THREAD

**Issue by splinter21** [Berkeley-Speech-Group/RT-VC#1]

**Comment by Louis0324** [comment]
The paper is still under review for ACL demo track. Once it is published, the paper link will be uploaded to the repo and demo page. Thank you for your interest!

**Comment by hermanseu** [comment]
@Louis0324  Hi Louis0324, the demo is amazing. I tried to train the models.  About the pitch model, we need to convert the pitch logits to pitch.  I just copy the code from  [pitch_tracker/model.py](https://github.com/Berkeley-Speech-Group/RT-VC/blob/main/src/pitch_tracker/model.py#L282), and write the forward func as follows:
```
 class Pitch_Wrapper(nn.Module):
    def __init__(self, pitch, config):
        super().__init__()
        self.pitch_model = pitch
        self.config = config

    def forward(self, mels):

        mels = mels.transpose(2,1)
        out_dict = self.pitch_model(mels)
        logits = out_dict['logits'] # [B, t, nbins]
        probs = torch.sigmoid(logits) # [B, t, nbins]
        periodicity = out_dict['periodicity'] # [B, 1, t]
        loudness = out_dict['loudness'] # [B, 1, t]

        weighted_sum = []
        L = 9
        for b in range (probs.shape[0]):
            prob = probs[b] # [t, nbins]
            center = torch.argmax(prob, dim=-1) # [t, ]
            start = torch.clamp(center-4, min=0) # [t, ]
            indices = torch.arange(L, device=mels.device).unsqueeze(0) + start.unsqueeze(1)
            indices = torch.clamp(indices, max=self.config['nbins']-1) # [t, L]
            seg_prob = prob[torch.arange(prob.shape[0], device=mels.device).unsqueeze(1), indices] # [t, L]
            s = ((indices * seg_prob).sum(dim=-1) / seg_prob.sum(dim=-1)).unsqueeze(0) # [1, t]
            weighted_sum.append(s)

        weighted_sum = torch.stack(weighted_sum, dim=0) # [B, 1, t]
        pitch = dequantize(weighted_sum, self.config['nbins'], self.config['fmin'], self.config['fmax'], self.config['f_base']) # [B, 1, t]
        return {
            'pitch': pitch,
            'periodicity': periodicity,
            'loudness': loudness
        }
```
And then, I get the l1 loss of pitch, periodicity and loudness.  The loss of pitch does not converge.  It looks like that pitch logits converting can not transfer gradient. How to deal this when you train the pitch model, can you give me some advice?

**Comment by Louis0324** [comment]
@hermanseu Hi, thank you for your interest. The training of the pitch tracker is largely based on [RMVPE](https://arxiv.org/abs/2306.15412) and [CREPE](https://arxiv.org/abs/1802.06182), I encourage you to take a look.

**Comment by hermanseu** [comment]
OK, thank you!

**Comment by hermanseu** [comment]
@Louis0324   How to get the pitch target label?
In RMVPE repo, the follow code likes to get pitch target label:
```
                    freq = 440 * (2.0 ** ((float(line) - 69.0) / 12.0))        # float(line) is the target pitch value ?
                    cent = 1200 * np.log2(freq/10)
                    index = int(round((cent-CONST)/20))
                    pitch_label[i][index] = 1
                    voice_label[i] = 1
```
the code will get index of [0, 360),  when using torchcrepe to get pitch,  I get index bigger than 360. 
So how you get the target pitch, and get the pitch label in [0, 256), as the config(pitch_tracker/yamls/rtpt.yaml) defines 256 bins?

**Comment by hermanseu** [comment]
Hi,  @Louis0324.   

I have trained all the models, but the outputs are not as good as expected.  Some can get normal voice, and some are very noisy.  About timbre, all the output are not same with the vc target. Just like the spk_emb does not work, and noly adjusted pitch.  
PS: When training the spk_encoder and vocoder models, I use one audio for vocoder and another audio of the same spk to get spk_emb.

Can you show me some details about training? How many hours data(how many spks) did you use to the models? Did you train the models separately, or in end_to_end manner? Are the yaml config files in the repo used by the demo?  Sorry about so many questions!

**Comment by Louis0324** [comment]
Please check our paper here: [https://arxiv.org/abs/2506.10289](https://arxiv.org/abs/2506.10289)
